cmd/kubeadm/app/cmd/phases/upgrade/node/kubeletconfig.go

```diff
@@ -75,24 +73,20 @@ func runKubeletConfigPhase() func(c workflow.RunData) error {
 			return err
 		}
 
-		// Gets the target kubelet version.
-		// by default kubelet version is expected to be equal to ClusterConfiguration.KubernetesVersion, but
-		// users can specify a different kubelet version (this is a legacy of the original implementation
-		// of `kubeam upgrade node config` which we are preserving in order to don't break GA contract)
-		kubeletVersionStr := cfg.ClusterConfiguration.KubernetesVersion
-		if data.KubeletVersion() != "" && data.KubeletVersion() != kubeletVersionStr {
-			kubeletVersionStr = data.KubeletVersion()
-			fmt.Printf("[upgrade] Using kubelet config version %s, while kubernetes-version is %s\n", kubeletVersionStr, cfg.ClusterConfiguration.KubernetesVersion)
-		}
+		// TODO: Checkpoint the current configuration first so that if something goes wrong it can be recovered
 
-		// Parse the desired kubelet version
-		kubeletVersion, err := version.ParseSemantic(kubeletVersionStr)
-		if err != nil {
-			return err
-		}
+		// Store the kubelet component configuration.
+		// By default the kubelet version is expected to be equal to cfg.ClusterConfiguration.KubernetesVersion, but
+		// users can specify a different kubelet version (this is a legacy of the original implementation
+		// of `kubeadm upgrade node config` which we are preserving in order to not break the GA contract)
```

Comment:
we are removing the command here:
https://github.com/kubernetes/kubernetes/pull/87975
the PR is lgtm/approved.

after it merges, can we download based on the ClusterConfiguration version only?

also, we should stop versioning the kubelet ConfigMap name, at some point:
xref https://github.com/kubernetes/kubeadm/issues/1582

Reply:
I'm +1 to preserve the logic as it is, I think it is more resilient

Reply:
Unfortunately, we can't remove it now. We have to keep it because `--kubelet-version` continues to be exposed through `kubeadm upgrade node` and it continues to work much in the same way as it did when `kubeadm upgrade node config` was around.
We didn't explicitly deprecate `--kubelet-version` until #87942 which was last week.
This is basically the last remaining vestige of the `kubelet.DownloadConfig` function in kubeadm, but we have to keep it for another year. That's the GA rules.